app/jobs/auth_trail/geocode_job.rb

```diff
@@ -3,19 +3,24 @@ class GeocodeJob < ApplicationJob
     def perform(login_activity)
       result =
         begin
-          Geocoder.search(login_activity.ip).first.try(:data)
+          Geocoder.search(login_activity.ip).first
         rescue => e
           Rails.logger.info "Geocode failed: #{e.message}"
           nil
         end
 
       if result
         login_activity.update!(
-          city: result["city"].presence,
-          region: result["region_name"].presence,
-          country: result["country_name"].presence
+          city: get(result, :city),
```

Comment:
Let's use `try` instead of adding a new method.

Reply:
Good call. :)